plugins/Monolog/config/config.php

```diff
@@ -97,6 +99,13 @@
     'Piwik\Plugins\Monolog\Handler\FileHandler' => DI\create()
         ->constructor(DI\get('log.file.filename'), DI\get('log.level.file'))
         ->method('setFormatter', DI\get('log.lineMessageFormatter.file')),
+    
+    '\Monolog\Handler\ErrorLogHandler' => DI\create()
+        ->constructor(null, DI\get('log.level.errorlog'))
```

Comment:
Looking at the ErrorHandler constructor, it seems like this shouldn't be set to null. If you want to only set the level parameter, and use defaults for the rest, you can use:

```
->constructorParameter('level', DI\get('log.level.errorlog'))
```

Reply:
That is one nifty dependency injection system you've got going!  So would it be like this?
```
   '\Monolog\Handler\ErrorLogHandler' => DI\create()
   ->constructorParameter('level', DI\get('log.level.errorlog'))
```

Other uses of constructorParameter in the code use \DI\autowire() or \DI\object().

Thanks so much for the guidance dizzy!

Reply:
@mwithheld yes, that looks correct! And yes it is definitely a nifty DI system! The library is called php-di and the creator, @/mnapoli, worked w/ us for a while so our integration works rather well.